tests/course/test_for_courses.py

```diff
@@ -0,0 +1,181 @@
+
+    @pytest.mark.django_db()
+    def test_delete_ref_member0(self, studentCourses0):
+        self.test_persist_student_course(studentCourses0)
+        studentCourses0.teacher_course_id.delete()
+        assert studentCourses0 not in StudentCourses.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_ref_member1(self, studentCourses0):
+        self.test_persist_student_course(studentCourses0)
+        studentCourses0.student_id.delete()
+        assert studentCourses0 not in StudentCourses.objects.all()
+
+    # CHECK FILTERS AND METHODS
+
+    def test_change_to_confirmed(self, studentCourses1):
+        assert studentCourses1.status == PENDING
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1.status == CONFIRMED
+
+    def test_teacher_pending(self, studentCourses1, studentCourses2, user0, user1):
+        # USER0 AS TEACHER
+        assert studentCourses1.teacher_course_id.course_id in StudentCourses.objects.get_teacher_pending(user0.pk)
+        assert studentCourses2.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user0.pk)
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user0.pk)
+        # USER1 AS TEACHER
+        assert studentCourses2.teacher_course_id.course_id in StudentCourses.objects.get_teacher_pending(user1.pk)
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user1.pk)
+        studentCourses2.change_to_confirmed()
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user1.pk)
+
+    def test_student_pending(sself, studentCourses1, studentCourses2, user0, user1):
+        # USER0 AS STUDENT
+        assert studentCourses1 not in StudentCourses.objects.get_student_pending(user0)
+        assert studentCourses2 in StudentCourses.objects.get_student_pending(user0)
+        studentCourses2.change_to_confirmed()
+        assert studentCourses2 not in StudentCourses.objects.get_student_pending(user0)
+        # USER1 AS STUDENT
+        assert studentCourses1 in StudentCourses.objects.get_student_pending(user1)
+        assert studentCourses2 not in StudentCourses.objects.get_student_pending(user1)
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1 not in StudentCourses.objects.get_student_pending(user1)
+
+    def test_student_courses(self, studentCourses1, studentCourses2, user0, user1):
+        # USER0 AS STUDENT
+        assert studentCourses1 not in StudentCourses.objects.get_student_courses(user0)
+        assert studentCourses2 in StudentCourses.objects.get_student_courses(user0)
+        # USER1 AS STUDENT
+        assert studentCourses1 in StudentCourses.objects.get_student_courses(user1)
+        assert studentCourses2 not in StudentCourses.objects.get_student_courses(user1)
```

Comment:
honestly, those tests are quite confusing. I'll suggest separating them and instead of comments have a good name for the test that explains what they test.